Arm/Insts/Common.lean

```diff
@@ -219,20 +219,32 @@ def CheckSPAlignment (s : ArmState) : Prop :=
 instance : Decidable (CheckSPAlignment s) := by unfold CheckSPAlignment; infer_instance
 
 @[state_simp_rules]
-theorem CheckSPAligment_of_w_different (h : StateField.GPR 31#5 ≠ fld) :
+theorem CheckSPAligment_w_different_eq (h : StateField.GPR 31#5 ≠ fld) :
   CheckSPAlignment (w fld v s) = CheckSPAlignment s := by
   simp_all only [CheckSPAlignment, state_simp_rules, minimal_theory, bitvec_rules]
 
+theorem CheckSPAligment_w_of_ne_sp_of (h : StateField.GPR 31#5 ≠ fld) :
```

Comment:
At first glance, theorems like `CheckAlignment*_sp_of` seem unnecessary to have, esp. in the face of those like `CheckSPAligment_*_eq`. Mention that they are useful to construct a non-effects proof for SP and point to `AxEffects`.

Reply:
I've added a comment